### PR TITLE
Use legal links config

### PR DESCRIPTION
**Pull request by mholthausen**
This PR lets the map client use the configuration of the legal links for the footer, when available. If this setting is not specified for a client, the already existing defined links are used as default.  
Needs https://github.com/terrestris/shogun/pull/642 and https://github.com/terrestris/shogun-util/pull/254 first be merged.

Also included is some code styling by eslint.

@terrestris/devs please review

### CODE REVIEW

File: src/store/legal/index.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  createSlice, PayloadAction
+} from '@reduxjs/toolkit';
+
+export type Legal = {
+  contact?: string;
+  imprint?: string;
+  privacy?: string;
+};
+
+const initialState: Legal = {};
```
Comment (dnlkoch) [inline src/store/legal/index.ts:11]: We might want to set the defaults here.

File: src/components/Footer/index.tsx [hunk 2]
```diff
@@ -94,37 +102,19 @@ export const Footer: React.FC<FooterProps> = ({
...
-          />
+          <Divider type="vertical" />
         </div>
-        <div
-          className="mouse-position-wrapper"
-        >
+        <div className="mouse-position-wrapper">
           {t('Footer.mousePosition')}:&nbsp;
           <div
             id="mouse-position"
```
Comment (dnlkoch) [inline src/components/Footer/index.tsx:120]: I would favour the previously adapted code style, but this is non-blocking.

File: src/store/legal/index.ts (hunk 1 above)
Comment (mholthausen) [reply src/store/legal/index.ts]: Defaults got moved from the component to the store.

File: src/components/Footer/index.tsx [hunk 3]
```diff
@@ -1,44 +1,40 @@
...
-  useTranslation
-} from 'react-i18next';
+import { useTranslation } from 'react-i18next';
 
 import ScaleCombo from '@terrestris/react-geo/dist/Field/ScaleCombo/ScaleCombo';
 import useMap from '@terrestris/react-geo/dist/Hook/useMap';
 
 import './index.less';
+import useAppSelector from '../../hooks/useAppSelector';
+import { Legal } from '../../store/legal';
```
Comment (simonseyock) [inline src/components/Footer/index.tsx:18]: This looks like it could have linting issues, please check,

Comment (mholthausen) [reply src/components/Footer/index.tsx:18]: Checked it locally and all checks here run successfully. `.eslintrc.js` says curly new line warning starts at two properties. VS Code edited the entries with just one property automatically on save

**Pull request merged**